.github/workflows/cloud-deploy/cloud-deploy-api.yml

```diff
@@ -0,0 +1,139 @@
+name: Deploy PROD API
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  workflow_call:
+
+jobs:
+
+  build_prod_image:
+    runs-on: ubuntu-latest
+    needs: test_api
+    timeout-minutes: 80
+    environment: Production
+    strategy:
+      matrix:
+        name: [ 'novu/api-ee' ]
+    outputs:
+      docker_image: ${{ steps.build-image.outputs.IMAGE }}
+    permissions:
+      contents: read
+      packages: write
+      deployments: write
+      id-token: write
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: ${{ contains (matrix.name,'-ee') }}
+          token: ${{ secrets.SUBMODULES_TOKEN }}
+      - uses: ./.github/actions/setup-project
+        with:
+          submodules: ${{ contains (matrix.name,'-ee') }}
+
+      - name: build api
+        run: pnpm build:api --skip-nx-cache
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
✏ **suggestion:** Can we remove the skip on NX Cache for `pnpm build:api` and `pnpm build:worker`? We can benefit greatly in build speed via cache usage from the pre-deploy phase‏

Reply:
```suggestion
        run: pnpm build:api
```